pkg/controllers/testing/table.go

```diff
@@ -214,27 +183,32 @@ func (tc *Testcase) Test(t *testing.T, scheme *runtime.Scheme, factory Reconcile
 		}
 	}
 
-	for i, exp := range tc.ExpectStatusUpdates {
-		if i >= len(clientWrapper.statusUpdateActions) {
-			t.Errorf("Missing status update: %#v", exp.Get())
+	compareActions(t, "status update", tc.ExpectStatusUpdates, clientWrapper.statusUpdateActions, statusSubresourceOnly, ignoreLastTransitionTime, safeDeployDiff, cmpopts.EquateEmpty())
+
+	// Validate the given objects are not mutated by reconciliation
+	if diff := cmp.Diff(originalGivenObjects, givenObjects, safeDeployDiff, cmpopts.EquateEmpty()); diff != "" {
+		t.Errorf("Given objects mutated by test %s (-expected, +actual): %v", tc.Name, diff)
+	}
+}
+
+func compareActions(t *testing.T, actionName string, expectedActionFactories []Factory, actualActions []SingularAction, diffOptions ...cmp.Option) {
```

Comment:
```suggestion
func compareActions(t *testing.T, actionName string, expectedActionFactories []Factory, actualActions []objectAction, diffOptions ...cmp.Option) {
```

https://github.com/projectriff/system/blob/873427cb3b598a0159d21e1e934ddb89ef63b5da/pkg/controllers/testing/client.go#L309-L312

Reply:
Adopted.